src/components/detail/chart/BarChart.vue

```diff
@@ -51,41 +54,37 @@ export default {
                     'transform', `translate(${margin.left},${margin.top})`
                   )
 
-            const data = self.data.detail ?
-                self.data : { detail: [] }
-
             function resize () {
                 const n = root.node(),
                       width = n.offsetWidth - margin.left - margin.right,
                       height = n.offsetHeight - margin.top - margin.bottom - padding,
                       x = scales.scaleTime().rangeRound([0, width]),
                       y = scales.scaleLinear().rangeRound([height, 0]),
                       xW = scales.scaleBand().align(0).padding(0.92),
-                      dates = data.detail.map((d) => new Date(Date.parse(d.month)))
+                      dates = detail.map((d) => new Date(Date.parse(d.month)))
 
                 x.domain(arr.extent(dates))
-                y.domain([0, arr.max(data.detail.map((d) => d.total))])
+                y.domain([0, arr.max(detail.map((d) => d.total))])
 
                 xW.range(x.range()).domain(x.domain())
 
                 svg.attr('width', n.offsetWidth).attr('height', n.offsetHeight)
                 g.attr('width', width).attr('height', height)
 
-                if (self.breakdown) {
-                    // this should be dynamic but we only have four colors
-                    // the data structure is bad, redo that in real code
-                    g.append('g').selectAll('.bar').data(data.detail)
+                if (typeof detail[0].total !=  'number') {
+                    // TODO max should only take into account the active breakdowns, not all
+                    const max = _.max(detail.map((r) => _.max(_.map(r.total, (d) => d))))
+                    y.domain([0, max])
+                    g.append('g').selectAll('.bar').data(detail)
                         .enter().selectAll('.minibar').data(function (d) {
                             // this should be passed in
```

Comment:
When you see them, make this and other "this should" type statements into TODOs so we can find them easily later.  Sorry I didn't do that in the prototype.